config/routes.rb

```diff
@@ -45,6 +45,8 @@ def matches?(request)
     get "claims/confirmation", as: :claim_confirmation, to: "submissions#show"
 
     get "timeout", to: "claims#timeout", as: :timeout_claim
+    get "existing-session", as: :existing_session, to: "claims#existing_session"
+    get "clear-session-and-redirect", to: "claims#clear_session_and_redirect", as: :clear_session_and_redirect
```

Comment:
I thought I had asked this in my first review, but couldn't find it - how do you feel about using a `GET` for a destructive action? Might a `POST` be more appropriate?

Reply:
Yeah, I wondered about using `DELETE` at first, but discounted it, so just switch to `GET`, but yeah, `POST` is a good shout

Reply:
@pezholio @lawrence-dxw Using `method: :post` looks to rely on JavaScript - is this something we need to run past accessibility?

Reply:
Hmmm - good shout. Maybe `button_to` would be a better choice, which would then generate a form

Reply:
> Yeah, I wondered about using `DELETE` at first, but discounted it, so just switch to `GET`, but yeah, `POST` is a good shout

What is it about `DELETE` that made you discount it?

Reply:
It's not acting on a specific resource, so didn't really make sense.

Reply:
I'm just asking in the design system Slack as to if they have any prior art on styling buttons as links, since then `button_to` would solve it neatly.

`DELETE` means we'd be stuck relying on JavaScript as well, since we can't `DELETE` from a standard form.

Reply:
It is acting on a singular resource, i.e. the current session.

Pretty sure Rails sets a `_method` param set to `DELETE` and under the hood sets up routing so that it will all work as you'd expect, even without JS.

Reply:
Some styling thoughts at https://github.com/alphagov/govuk-design-system-backlog/issues/159. One from Slack which I don't completely hate is the idea of moving all the buttons outside of the interruption dialog (although we should still really have them on the same line as per the [design system](https://design-system.service.gov.uk/components/button/).

![localhost_3000_student-loans_existing-session](https://user-images.githubusercontent.com/619082/72262363-b6a61080-360e-11ea-9f55-f852513aeb92.png)

It looks like `_method` is some magic which is [added by JS at runtime](https://edgeguides.rubyonrails.org/working_with_javascript_in_rails.html#data-method) vs when the view is rendered. Disabling JS means that the link just performs a `GET` (and 404s).

`DELETE` would make sense if we reworked the URLs so that it was `DELETE`ing something like `student-loans/session`, but that still runs into the 'needs JS' problem.

Reply:
Ah I meant was if we use a form, it will just work. Rails will automatically set a hidden input like this: 

```ruby
<input type="hidden" value="delete" name="_method">
```

which will route correctly without JS.

Reply:
So suddenly we're reeaaaaally close to having an actual RESTful resource for the session... What would this code look like if the actions were on the existing sessions controller, rather than the claims controller?

Reply:
That's what I had originally, but it was becoming problematic as there were a whole bunch of helpers etc that I needed for the views, and it didn't seem right to inherit the sessions controller from the base controller

Reply:
I think it would make sense to move `#clear_session_and_redirect` to the sessions controller since that's the domain it's acting on. I'm less certain about `#existing-session`, since it's the _claim_ which is informing the user it cannot continue.

I am, however, not entirely sold on this view.

Reply:
I think that makes sense. So move the action to clear the session to the sessions controller and keep the existing action on claims?

Reply:
That'll work. Then a `DELETE` makes much more sense

Reply:
I remember why this was a minor pain - we kept it in the claims flow so that:

1. It could use `BasePublicController#clear_claim_session`, and
2. We didn't have to pass the target policy into the controller (which is needed to fulfil the redirect), although I guess it's not too much faff to just sling a query parameter along like `?redirect_to_policy`

Reply:
1. Reckon `SessionsController` should probably be inheriting from `BasePublicController` anyway. It may or may not mean skipping some `before_action`s...
2. Yep, I reckon make it as much a RESTful `destroy` action as possible to lean on conventions. A query param for the redirect sounds like a good shout. I wondered if a more generic `redirect_to` param that accepts a path would make sense, but actually it's probably safer to expect a policy name and always redirect to the start of claim form as you suggest.